src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include <boost/algorithm/string.hpp>
+#include "ReportCu.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include "core/common/utils.h"
+
+namespace qr = xrt_core::query;
+
+boost::property_tree::ptree
+parse_cu_status(uint32_t cu_status)
+{
+  boost::property_tree::ptree pt;
+  std::vector<std::string> result;
+
+  std::string cu_stat_str = xrt_core::utils::parse_cu_status(cu_status);
+  std::string t_str = cu_stat_str.substr(1, cu_stat_str.size()-2);
+  boost::split(result, t_str, boost::is_any_of("|"));
+
+  pt.put("bit_mask",	boost::str(boost::format("0x%x") % cu_status));
+  boost::property_tree::ptree ptSt_arr;
+  for (int i = 0; i < (int)result.size(); i++)
+    ptSt_arr.push_back(std::make_pair("", boost::property_tree::ptree(result[i])));
+
+  pt.add_child( std::string("bits_set"), ptSt_arr);
+
+  return pt;
+}
+
+boost::property_tree::ptree
+populate_cus(const xrt_core::device *device)
+{
+  boost::property_tree::ptree pt;
+  std::vector<char> ip_buf;
+  std::vector<std::tuple<uint64_t, uint32_t, uint32_t>> cu_stats;
+
+  try {
+    ip_buf = xrt_core::device_query<qr::ip_layout_raw>(device);
+    cu_stats = xrt_core::device_query<qr::kds_cu_info>(device);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+  }
+
+  if(ip_buf.empty() || cu_stats.empty())
+    return pt;

```

Comment:
If there is no buffer or status, should there be an "error_msg" produced indicating so instead of "failing" quietly.